src/components/capture/AutoDiscoverySettings.tsx

```diff
@@ -0,0 +1,224 @@
+import {
+    FormControl,
+    FormControlLabel,
+    Stack,
+    Switch,
+    Typography,
+} from '@mui/material';
+import useAutoDiscovery from 'components/capture/useAutoDiscovery';
+import { useEditorStore_queryResponse_draftSpecs } from 'components/editor/Store/hooks';
+import { isEqual, isObject } from 'lodash';
+import { useEffect, useMemo } from 'react';
+import { FormattedMessage } from 'react-intl';
+import {
+    useFormStateStore_isActive,
+    useFormStateStore_setFormState,
+} from 'stores/FormState/hooks';
+import { FormStatus } from 'stores/FormState/types';
+import {
+    useSchemaEvolution_addNewBindings,
+    useSchemaEvolution_autoDiscover,
+    useSchemaEvolution_evolveIncompatibleCollections,
+    useSchemaEvolution_setAddNewBindings,
+    useSchemaEvolution_setAutoDiscover,
+    useSchemaEvolution_setEvolveIncompatibleCollections,
+    useSchemaEvolution_setSettingsSaving,
+    useSchemaEvolution_settingsSaving,
+} from 'stores/SchemaEvolution/hooks';
+
+interface Props {
+    readOnly?: boolean;
+}
+
+function AutoDiscoverySettings({ readOnly }: Props) {
+    // Draft Editor Store
+    const draftSpecs = useEditorStore_queryResponse_draftSpecs();
+
+    // Form State Store
+    const formActive = useFormStateStore_isActive();
+    const setFormState = useFormStateStore_setFormState();
+
+    // Schema Evolution Store
+    const autoDiscover = useSchemaEvolution_autoDiscover();
+    const setAutoDiscover = useSchemaEvolution_setAutoDiscover();
+
+    const addNewBindings = useSchemaEvolution_addNewBindings();
+    const setAddNewBindings = useSchemaEvolution_setAddNewBindings();
+
+    const evolveIncompatibleCollections =
+        useSchemaEvolution_evolveIncompatibleCollections();
+    const setEvolveIncompatibleCollections =
+        useSchemaEvolution_setEvolveIncompatibleCollections();
+
+    const settingsSaving = useSchemaEvolution_settingsSaving();
+    const setSettingsSaving = useSchemaEvolution_setSettingsSaving();
+
+    const updateAutoDiscoverySettings = useAutoDiscovery();
+
+    const settingsExist = useMemo(
+        () =>
+            draftSpecs.length > 0 &&
+            Object.hasOwn(draftSpecs[0].spec, 'autoDiscover') &&
+            isObject(draftSpecs[0].spec.autoDiscover),
```

Comment:
`isObject` should handle a missing object fine so we should remove the `hasOwn' check and just use `isObject` with an optional chain.